pywr/parameters/_control_curves.pyx

```diff
@@ -205,14 +208,27 @@ cdef class ControlCurveInterpolatedParameter(BaseControlCurveParameter):
     >>> cost = ControlCurveInterpolatedParameter(storage_node, ccs, values)
     >>> storage_node.cost = cost
     """
-    def __init__(self, model, storage_node, control_curves, values):
-        super(ControlCurveInterpolatedParameter, self).__init__(model, storage_node, control_curves)
+    def __init__(self, model, storage_node, control_curves, values=None, parameters=None, **kwargs):
+        super(ControlCurveInterpolatedParameter, self).__init__(model, storage_node, control_curves, **kwargs)
         # Expected number of values is number of control curves plus two.
         nvalues = len(self.control_curves) + 2
-        if len(values) != nvalues:
-            raise ValueError('Length of values should be two more than the number of '
-                             'control curves ({}).'.format(nvalues))
-        self.values = values
+        self.parameters = None
+
+        if values is not None:
+            if len(values) != nvalues:
+                raise ValueError('Length of values should be two more than the number of '
+                                 'control curves ({}).'.format(nvalues))
+            self.values = np.asarray(values)
+
+        elif parameters is not None:
+            if len(parameters) != nvalues:
+                raise ValueError('Length of parameters should be two more than the number of '
+                                 'control curves ({}).'.format(nvalues))
+            self.parameters = list(parameters)
+            for p in self.parameters:
```

Comment:
Add comment explaining what this is for.

Reply:
I've added this comment. I also added it to the `ControlCurveParameter` init method too.